setup.cfg

```diff
@@ -38,7 +38,7 @@ dev =
     pytest-cov>=3.0.0,<5
     pytest_mock>=3.8.0,<4
     towncrier
-    ruff
+    ruff>=0.12,<0.13
```

Comment:
Pinned to `<0.13` to get the CI green, I can fix changes in ruff 0.13 in separate PR.